wordpress-mobile/WordPress-Android
# Issue/11151 insert preview bottom bar

Fixes #11151

Review Instructions:
1. Review this PR - I recommend reviewing it by commit instead of comparing all the changes at once
2. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/11611 is already merged
3. Make sure `master-custom-image-picker-in-gutenberg` is merged into develop
4. Merge `develop` into `feature/media-editing-phase-two` 
5. Merge `feature/media-editing-phase-two` into this branch
6. Change the target branch to `feature/media-editing-phase-two`
7. Merge this PR

(Doesn't really matter who performs the steps 4-6. Feel free to ping me and I can do it. Thanks)

Changes:
- Added a bottom bar with "Preview/Insert" actions when the media picker is started from gutenberg
- Hid "Done" action when the new bottom bar is shown
- Refactored the (show/hide/isShowing)BottomBar methods

Remaining:
- [ ] Implement OnClickListener for the Preview button

@mbshakti Could you please review these changes? Thanks!
- Since the picker already shows the number of selected items in the toolbar (when multiselection is enabled) I decided not to use `Insert XY` button and use just `Insert`. Is that ok?
- I've used some simply hide/show animations on the bottom bars and it looks imo ok. Wdyt?


Update: I've replaced "Preview" button with "Edit" button. I didn't update the gifs as it seems unnecessary.

| BEFORE | AFTER |
|---|---|
| ![old-preview-insert-bar](https://user-images.githubusercontent.com/2261188/79116914-f3a6ae00-7d89-11ea-8c78-cfac052e3a19.gif) | ![new-preview-insert-bar](https://user-images.githubusercontent.com/2261188/79116806-b3473000-7d89-11ea-94f5-61f104487d65.gif) |
| ![old-preview-insert-bar-gallery](https://user-images.githubusercontent.com/2261188/79116919-f6090800-7d89-11ea-8c2c-fb81655f7731.gif) | ![new-preview-insert-bar-gallery](https://user-images.githubusercontent.com/2261188/79116819-bc380180-7d89-11ea-9aa6-a8ebdf216bad.gif) |

To test:
1. Create a new post in gutenberg
2. Add an image block
3. Select "Choose from device"
4. Select an image and notice a bottom bar with Preview/Insert is shown (make sure the "checkmark" done action in the toolbar is not visible)
5. Click on the same image and notice the bar disappears
6. Repeat steps 1-5 with a gallery block and make sure multi-selection works as expected

-------------------------------

1. Create a new post in Aztec
2. Add an image and make sure the custom media picker still looks the same as in `develop`

----------------------------------

1. My Site
2. Change Gravatar
3. Make sure the media picker looks the same as in `develop`

----------------------------------

1. My Site
2. Click on the avatar in the toolbar (me section)
3. Change Avatar
4. Make sure the media picker looks the same as in `develop`



PR submission checklist:
